### ISSUE TITLE
CI C pipeline error rtrim expects parameter 1 to be string, bool given

### ISSUE THREAD

**Issue by Jaimin-Devhero** [composer/composer#10670]
I have set up GitLab's CI CD pipeline in my project.

I am facing issues like the below error ("rtrim() expects parameter 1 to be string, bool given") when trying to run the composer install command from GitLab's ci cd pipeline job.
But when I run that command manually using ssh then works fine.

Please help me with this guys.

below is the error stash:

**Comment by Seldaek** [comment]
What Composer version are you running?

I can't make sense of why this would fail with latest code.. but yet it seems to fail.

If you can also make sure you run it with `-vvv` and share full output of the bootstrap it may hint at what the problem is.

**Comment by Jaimin-Devhero** [comment]
My Composer version is: **Composer 2.3.0 2022-03-30 11:15:36**

**Comment by tisalmin** [comment]
Same here, deploying to AWS and version is 2.3.0. Forcing deploy to use version 2.2.9 works.

**Comment by bensebborn** [comment]
Strangely the version pinning on install doesnt seem to work

` php -c /etc/php-cli.ini /tmp/composer --install-dir=/usr/local/bin --filename=composer --version=2.2.10
`

`[ec2-user@ip-10-0-202-128 ~]$ composer -V
Composer 2.3.0 2022-03-30 11:15:36
`

**Comment by clementmas** [comment]
Yes `COMPOSER_HOME` is set properly. I'm also now getting this error all of the sudden.

**Comment by bensebborn** [comment]
I don't explicity set that env var (Can't see that mentioned in install docs?) - Presume the install script sets that?

`php -c /etc/php-cli.ini /tmp/composer --install-dir=/usr/local/bin --filename=composer --version=2.2.10`

`printenv COMPOSER_HOME` 
outputs nothing after install script is run

**Comment by Seldaek** [comment]
https://github.com/composer/composer/releases/tag/2.3.1 is out now. Everyone please try release candidates next time :) If you update your scripts now you can change `--version=xxx` into `--preview` so you get the latest stable OR latest RC when available and you are ready to test the next one!

**Comment by bensebborn** [comment]
@Seldaek the --version doesn't seem to work but I'll open a new ticket. Thanks for fix

**Comment by clementmas** [comment]
That fixed it. Thanks for your reactivity @Seldaek!

I didn't know about the `--preview` flag.

**Comment by tisalmin** [comment]
> Sorry just to make sure I can clean this up correctly, do all of you have the `COMPOSER_HOME` env var set? Because otherwise it seems you should be getting an exception about HOME being unset (which is the issue triggering the rtrim problem).

Yes
